src/Mutator/FunctionSignature/ProtectedVisibility.php

```diff
@@ -49,6 +50,33 @@ public function shouldMutate(Node $node): bool
             return false;
         }
 
+        if ($this->hasSameProtectedParentMethod($node)) {
```

Comment:
Just a small thing, but maybe its slightly faster to first check if the node is protected? before checking if it has a protected parent.

And ofcourse the same goes for public